src/crimson/osd/shard_services.cc

```diff
@@ -93,12 +94,21 @@ seastar::future<> PerShardState::broadcast_map_to_pgs(
   return seastar::parallel_for_each(
     pgs.begin(), pgs.end(),
     [=, &shard_services](auto& pg) {
+      auto old_map = pg.second->get_osdmap();
       return shard_services.start_operation<PGAdvanceMap>(
 	pg.second,
 	shard_services,
 	epoch,
-	PeeringCtx{}, false).second;
-    });
+	PeeringCtx{}, false).second.then(
+     [this, old_map, epoch, &shard_services, &pg] {
+      return shard_services.start_operation<PGSplitting>(
```

Comment:
Note: We can check/identify splits before starting the operation. For now, I think the logic separation is preferable over the slight overhead (on non-splits advances).